sympy/solvers/tests/test_diophantine.py

```diff
@@ -583,6 +582,58 @@ def check_solutions(eq):
 
     return okay
 
+
+def check_linear(eq, param=symbols("t", Integer=True)):
+    """
+    Determines whether solutions returned by diop_solve() satisfy the
+    original equation.
+    """
+
+    eq = eq.expand(force=True)
```

Comment:
I copied the first few lines of classify_diop().  I didn't check whether force was still necessary.